torch/distributions/gumbel.py

```diff
@@ -0,0 +1,54 @@
+from numbers import Number
+
+import math
+
+import torch
+from torch.distributions import constraints
+from torch.distributions.distribution import Distribution
+from torch.distributions.utils import broadcast_all
+
+
+class Gumbel(Distribution):
+    r"""
+    Samples from a Gumbel Distribution.
+
+    Examples::
+
+        >>> m = Gumbel(torch.Tensor([1.0]), torch.Tensor([2.0]))
+        >>> m.sample()  # sample from Gumbel distribution with loc=1, scale=2
+         1.0124
+        [torch.FloatTensor of size 1]
+
+    Args:
+        loc (float or Tensor or Variable): Location parameter of the distribution
+        scale (float or Tensor or Variable): Scale parameter of the distribution
+    """
+    has_rsample = True
+    params = {'loc': constraints.real, 'scale': constraints.positive}
+
+    def __init__(self, loc, scale):
+        self.loc, self.scale = broadcast_all(loc, scale)
+        if isinstance(loc, Number) and isinstance(scale, Number):
+            batch_shape = torch.Size()
+        else:
+            batch_shape = self.scale.size()
+        super(Gumbel, self).__init__(batch_shape)
+
+    @constraints.dependent_property
```

Comment:
This constraint is not dependent on parameters, so instead you should simplify and move this up to
```py
params = ...
support = constraints.real
```